firebase-perf/src/main/java/com/google/firebase/perf/application/AppStateMonitor.java

```diff
@@ -308,7 +308,7 @@ private void sendScreenTrace(Activity activity) {
     int slowFrames = 0;
     int frozenFrames = 0;
     // Stops recording metrics for this Activity and returns the currently-collected metrics
-    SparseIntArray[] arr = frameMetricsAggregator.remove(activity);
+    SparseIntArray[] arr = frameMetricsAggregator.reset();
```

Comment:
This is the critical change for this PR.

Earlier we were stopping the aggregator for a particular activity. With reset, we don't do this per activity. So, concerned if there is any sideways impact for the app when doing reset.

Reply:
The sideway impact is that `frameMetricsAggregator` will be monitoring all the activities that are added even after `reset()` is called. However, from the perspective of measuring the correct screen performance, this is fine. The reason is that there will only be one activity alive at a moment. We start measuring the performance when the activity starts and stop measuring when the activity stops. When the activity stops, we will call `reset()` to clear the data in  `frameMetricsAggregator` so that the next activity will have clean data at the beginning.

Reply:
Discussed offline. Will add TODOs and comments.